dotCMS/src/main/java/com/dotcms/content/elasticsearch/util/ESMappingUtilHelper.java

```diff
@@ -0,0 +1,331 @@
+                            + fieldVariable.key();
+
+                    if (field != null) {
+                        message += ". Field: " + field.name();
+                    }
+
+                    if (type != null) {
+                        message += ". Content Type: " + type.name();
+                    }
+
+                    message += ". Custom mapping will be ignored for index: " + indexName;
+                    Logger.warn(ESMappingUtilHelper.class, message, e);
+                }
+            }
+        } catch (DotDataException e) {
+            Logger.warn(ESMappingUtilHelper.class,
+                    "Error setting custom index mapping for index " + indexName, e);
+        }
+        return mappedFields;
+    }
+
+    /**
+     *
+     * @param indexName Index where the mapping will be applied
+     * @param mappedFields Collection of fields already mapped in the index. This collection is used to avoid duplicate mappings for fields, which could cause an explosion
+     */
+    private void addMappingForRemainingFields(final String indexName,
+            final Set<String> mappedFields) {
+        try {
+            final List<ContentType> contentTypes = contentTypeAPI.findAll();
+            contentTypes.forEach(contentType -> contentType.fields().forEach(
+                    field -> addMappingForFieldIfNeeded(indexName, contentType, field,
+                            mappedFields)));
+        } catch (DotDataException e) {
+            Logger.warnAndDebug(ESMappingUtilHelper.class,
+                    "It was not possible to get content types to map field types in Elasticsearch"
+                            + indexName, e);
+        }
+    }
+
+    /**
+     * Defines an ES custom mapping for dates, numbers and text fields, excluding those that match the mapping defined in the `es-content.mapping.json` file
+     * @param indexName Index where the mapping will be applied
+     * @param contentType Content type's whose field will be mapped
+     * @param field Field to be mapped
+     * @param mappedFields Collection of fields already mapped in the index. This collection is used to avoid duplicate mappings for fields, which could cause an explosion
+     */
+    private void addMappingForFieldIfNeeded(final String indexName,
+            final ContentType contentType, final Field field, final Set<String> mappedFields) {
+        final String fieldVariableName = (contentType.variable() + StringPool.PERIOD + field.variable())
+                        .toLowerCase();
+        if (!mappedFields.contains(fieldVariableName)) {
+            String mappingForField = null;
+            if (field instanceof DateField || field instanceof DateTimeField
+                    || field instanceof TimeField) {
+                mappingForField = "\"type\":\"date\",\n";
+                mappingForField += "\"format\": \"yyyy-MM-dd't'HH:mm:ss||MMM d, yyyy h:mm:ss a||yyyy-MM-dd HH:mm:ss||yyyy-MM-dd||epoch_millis\"\n";
```

Comment:
Codacy found an issue: [Prefer StringBuilder (non-synchronized) or StringBuffer (synchronized) over += for concatenating strings](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5739222)